src/intervals/exact_literals.jl

```diff
@@ -0,0 +1,66 @@
+struct ExactReal{T <: Real} <: Real
+    value::T
+end
+
+# Promotion to Interval
+Base.promote_rule(::Type{Interval{T}}, ::Type{ExactReal{S}}) where {T<:NumTypes,S<:Real} =
+    Interval{promote_numtype(T, S)}
+
+Base.promote_rule(::Type{ExactReal{T}}, ::Type{Interval{S}}) where {T<:Real,S<:NumTypes} =
+    Interval{promote_numtype(T, S)}
+
+function Base.convert(::Type{Interval{T}}, x::ExactReal) where {T<:NumTypes}
+    y = interval(T, x)
+    return _unsafe_interval(bareinterval(y), decoration(y), true)
```

Comment:
This may be unnecessary. Doesn't `y` already satisfy `isguaranteed(y) == true` already?